openlibrary/plugins/openlibrary/stats.py

```diff
@@ -282,24 +278,27 @@ def increment_error_count(key):
 TEMPLATE_SYNTAX_ERROR_RE = re.compile(r"File '([^']+?)'")
 
 
-def find_topmost_useful_file(exception, tback):
+def find_topmost_useful_file(
+    exception: BaseException, tback: TracebackType | None
+) -> str:
     """
     Find the topmost path in the traceback stack that's useful to report.
 
     :param BaseException exception: error from e.g. sys.exc_inf()
     :param TracebackType tback: traceback from e.g. sys.exc_inf()
-    :rtype: basestring
     :return: full path
     """
     file_path = 'none'
+
     while tback is not None:
         cur_file = tback.tb_frame.f_code.co_filename
         if '/openlibrary' in cur_file:
             file_path = cur_file
         tback = tback.tb_next
 
     if file_path.endswith('template.py') and hasattr(exception, 'msg'):
-        m = TEMPLATE_SYNTAX_ERROR_RE.search(exception.msg)
+        # mypy is not picking up hasattr(exception, 'msg')
```

Comment:
This was a change between Python 2 and Python 3 where `Exception.msg` was removed.  In Py3, you need to extract msg as `Exception.args[1]` or you will raise a NameError inside your exception.  See https://portingguide.readthedocs.io/en/latest/exceptions.html#iterating-exceptions

Reply:
I hope I got this right.